nucliadb/nucliadb/ingest/orm/processor/__init__.py

```diff
@@ -452,12 +462,8 @@ async def txn(
                     await sequence_manager.set_last_seqid(txn, partition, seqid)
                 await txn.commit()
 
-                # Slug may have conflicts as its not partitioned properly. We make it as short as possible
-                # XXX Does this write need to happen every time?
-                async with self.driver.transaction() as slug_txn:
-                    resource.txn = slug_txn
-                    await resource.set_slug()
-                    await slug_txn.commit()
+                if created or resource.slug_modified:
+                    await self.commit_slug(resource)
```

Comment:
Is this function useful? Why not adding a `resource.set_slug()` inside this if before `txn.commit`? Makes sense to have two transactions?

Reply:
Yes, we want to commit the slug in a separate (and shorter) transaction on purpose. That was the previous logic and I prefer to keep it as is

IMO the function is useful for readability: we are doing more than one thing to commit the slug.